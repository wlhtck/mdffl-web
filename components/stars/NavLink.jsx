import React from 'react';
import styled from 'styled-components';
import { string } from 'prop-types';
import { white } from '../nebulas/colors';

const NavLink = ({ url, text, className }) => (
  <a href={url} className={className}>
    {text}
  </a>
);

NavLink.propTypes = {
  url: string.isRequired,
  text: string.isRequired,
  className: string,
};

NavLink.defaultProps = {
  className: '',
};

export default styled(NavLink)`
  color: ${white};
  font-family: 'Raleway', sans-serif;
  font-size: 16px;
  text-decoration: none;
`;
