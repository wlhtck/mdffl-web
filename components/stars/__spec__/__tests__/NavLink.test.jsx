import React from 'react';
import 'jest-styled-components';
import { render } from 'enzyme';
import NavLink from '../../NavLink';

describe('components/starts/NavLink', () => {
  it('renders a link', () => {
    expect(render(<NavLink href="/about">Button</NavLink>)).toMatchSnapshot();
  });
});
