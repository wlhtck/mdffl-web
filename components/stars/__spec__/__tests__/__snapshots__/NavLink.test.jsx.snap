// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/starts/NavLink renders a link 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FFFFFF;
  font-family: 'Raleway',sans-serif;
  font-weight: normal;
  font-size: 16px;
  line-height: normal;
  margin: 0 0 32px;
}

<a
  class="withStyles-u5urts-0 c0"
  href="/about"
>
  CTA
</a>
`;
